gfam/tasks/find_unassigned/task.py

```diff
@@ -53,22 +57,25 @@ def run(self, assignment_file: str, fasta_file: str,
         self.log.info("Getting unassigned pieces")
         self._print_unassigned()
 
-    def _process_sequences_file_old(self, fname: str) -> None:
-        """ Old version, all the entries loaded into memory
+    def _process_sequences(self, fname: str) -> None:
+        """Reads and processes a fasta file
+
+        Parameters
+        ----------
+        fname : str
+            path to a fasta file
         """
         self.log.info(f"Loading sequences from {fname}...")
         parser = fasta.Parser(open_anything(fname))
         parser = fasta.regexp_remapper(parser,
                                        self.sequence_id_regexp)
-        seqs, lens = [], []
+        seqcat_keys = set(self.seqcat.keys())
```

Comment:
This creates an unneccesary set that takes extra memory. You can just change line 74 into

`if seq.id not in self.seqcat and len(seq.seq) >= self.maximum:`

Reply:
Well spotted.. I just pushed that changes this!